frontend/src/components/forms/EditResident.tsx

```diff
@@ -0,0 +1,315 @@
+      setRoomNumber(parseInt(inputValue, 10));
+      setRoomNumberError(false);
+    }
+  };
+
+  const handleMoveInDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (moveOutDate && inputValue > moveOutDate) {
+        setMoveInDateError(true);
+      }
+      else {
+        setMoveInDate(inputValue);
+        setMoveInDateError(false);
+      }
+    }
+  };
+  
+  const handleMoveOutDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (inputValue >= moveInDate) {
+        setMoveOutDate(inputValue);
+        setMoveOutDateError(false);
+      }
+      else {
+        setMoveOutDateError(true);
+      }
+    } 
+  };
+
+  const handleBuildingChange = (
+    selectedOption: SingleValue<{ label: string; value: string }>,
+  ) => {
+    if (selectedOption !== null) {
+      setUserBuilding(selectedOption.value);
+      setBuildingError(false);
+    }
+  };
+
+  const handleSave = () => {
+    setInitialsError(initials.length !== 2);
+    setRoomNumberError(roomNumber.toString().length !== 3);
+    setBuildingError(userBuilding === "");
+
+    //  Prevents form submission if any required values are incorrect
+    if (
+      initials.length !== 2 ||
+      moveInDateError || 
+      roomNumberError ||
+      userBuilding === "" ||
+      moveOutDateError
+    ) {
+      return;
+    }
+    editRes();
+    setShowAlert(true);
+    toggleClose();
+  };
+
+  // Timer to remove alert
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  return (
+    <>
+      <Box>
+        <Modal isOpen={isOpen} onClose={toggleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>Edit Resident</ModalHeader>
+            <ModalCloseButton size="lg" />
+            <ModalBody>
+              <Divider />
+              <Row style={{ marginTop: "16px" }}>
+                <Col>
+                  <FormControl isRequired isInvalid={initialsError}>
+                    <FormLabel>Resident Initials</FormLabel>
+                    <Input
+                      defaultValue={initial}
+                      value={initials}
+                      onChange={handleInitialsChange}
+                    />
+                    <FormErrorMessage>
+                      Resident Initials are required and must contain 2 letters.
+                    </FormErrorMessage>
+                  </FormControl>
+                </Col>
+                <Col>
+                  <FormControl isRequired isInvalid={roomNumberError}>
+                    <FormLabel>Room Number</FormLabel>
+                    <Input
+                      value={roomNumber}
+                      onChange={handleRoomNumberChange}
+                      type="number"
+                    />
+                    <FormErrorMessage>
+                      Room Number is required and must only contain numbers.
```

Comment:
nit: same comment as above on the error messages, it might make it nicer if we chuck these into variables above